src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Color = `red` | `green` | `blue`;
+type Shape = `triangle` | `circle` | `rectangle`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const NotPositive = a <= 0 || b <= 0 || c <= 0;
+    const isNotTriangle = a + b <= c || a + c <= b || b + c <= a;
+
+    if (NotPositive || isNotTriangle) {
+      throw new Error('incorrect value');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message.
![image](https://github.com/mate-academy/ts_figures_classes/assets/63243136/7c3ad104-94d4-413d-b624-d87634234ae1)